### ISSUE TITLE
Option to specify stdin filename

### ISSUE THREAD

**Issue by lassik** [Unibeautify/unibeautify-cli#18]
Something that's really useful for editor plugins is an option to say what filename the contents of stdin are coming from.

* [prettier](https://github.com/prettier/prettier): this option is called `--stdin-filepath`
* [rufo](https://github.com/ruby-formatter/rufo): this option is called `--filename`

In my experience, it's easiest and most reliable to implement an editor plugin by not using disk files at all. Instead, just feed unformatted code to stdin and get the formatted code back from stdout. But even when using stdin/stdout, it's useful if we can know the filename of the code (the code usually comes from a disk file, but may have been modified in the text editor and the modifications have not been saved, so the disk file is different from the editor buffer that is sent to stdin).

The point of knowing the filename is that:

* Formatter error/warning messages can show a filename that is intuitive to the user.
* Unibeautify can find `.unibeautifyrc` and possibly other project settings files based on the filename.

**Comment by Glavin001** [comment]
> In my experience, it's easiest and most reliable to implement an editor plugin by not using disk files at all. 

💯% agree.

> The point of knowing the filename is that:

And these are both great points.

---

Unibeautify CLI should absolutely support passing the file name.
Applicable code: https://github.com/Unibeautify/unibeautify-cli/blob/master/src/cli.ts#L74-L78
Can pass `filePath` as documented: https://api.unibeautify.com/interfaces/_beautifier_.beautifydata.html

Pull Requests welcome!

**Comment by szeck87** [comment]
I'm incorporating this into the `ci-fixes` branch.  I originally meant to work on this issue, but had to get everything else working first.

**Comment by lassik** [comment]
> @lassik please give this a try with @unibeautify/cli@0.3.0

```
$ sudo npm install --global @unibeautify/cli@0.3.0/usr/local/bin/unibeautify -> /usr/local/lib/node_modules/@unibeautify/cli/dist/cli.js
+ @unibeautify/cli@0.3.0
updated 1 package in 3.644s

$ echo | unibeautify --language JavaScript
(node:20532) UnhandledPromiseRejectionWarning: NestedError: Could not require module '@unibeautify/beautifier-prettydiff'
    at requireg (/usr/local/lib/node_modules/@unibeautify/cli/node_modules/requireg/lib/requireg.js:14:11)
    at beautifierNames.map.beautifierName (/usr/local/lib/node_modules/@unibeautify/cli/dist/index.js:27:16)
    at Array.map (<anonymous>)
    at loadBeautifiers (/usr/local/lib/node_modules/@unibeautify/cli/dist/index.js:26:41)
    at findInstalledBeautifiers.then (/usr/local/lib/node_modules/@unibeautify/cli/dist/index.js:34:16)
Caused By: Error: Cannot find module 'prettydiff2'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
    at Function.Module._load (internal/modules/cjs/loader.js:520:25)
    at Module.require (internal/modules/cjs/loader.js:650:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@unibeautify/beautifier-prettydiff/dist/src/index.js:4:20)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
(node:20532) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:20532) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by szeck87** [comment]
Make sure you have `[unibeautify](https://www.npmjs.com/package/unibeautify)` installed globally as well.

**Comment by lassik** [comment]
Ok, now I can use the `--file-path` option 😄  But I wasn't able to test whether it loads `.unibeautifyrc*` from that file path's directory because of https://github.com/Unibeautify/beautifier-clang-format/issues/29.

At a glange, it seems that `filePath` and `options` are passed at the same time in https://github.com/Unibeautify/unibeautify-cli/blob/da0006c8f92c268159be8cf833a93a9f653fc58f/src/Runner.ts#L33 so `filePath` doesn't have the chance to affect where `options` are loaded from. Does Unibeautify core do its own options parsing based on `filePath`?

**Comment by szeck87** [comment]
Loading a .unibeautifyrc.yml file does not work yet.  When it will be supported, it will likely need to be passed in via the `-c` flag, but we'll see.

**Comment by lassik** [comment]
That's OK. I would suggest looking up `.unibeautifyrc*` based on the pathname of each beautified file by default (or from `--file-path` when given) since `clang-format`, `git` and countless other programs use a similar method. `--config-file` would override all of these, if given.

**Comment by szeck87** [comment]
Closed with #25
